lightning-persister/src/background_processor.rs

```diff
@@ -0,0 +1,219 @@
+use lightning::chain;
+use lightning::chain::chaininterface::{BroadcasterInterface, FeeEstimator};
+use lightning::chain::keysinterface::{ChannelKeys, KeysInterface};
+use lightning::ln::channelmanager;
+use lightning::ln::channelmanager::ChannelManager;
+use lightning::util::logger::Logger;
+use lightning::util::ser::Writeable;
+use std::sync::Arc;
+use std::sync::mpsc::{self, Sender, TryRecvError};
+use std::thread;
+use std::time::{Duration, SystemTime};
+
+/// BackgroundProcesser runs a background thread that monitors whether the ChannelManager it's given on
+/// start has new updates to persist. If it does, then it writes said ChannelManager to disk/backups.
+pub struct BackgroundProcessor {
+	#[cfg(test)]
+	pub thread_terminator: Sender<()>,
```

Comment:
Should we just use an AtomicBool for this? If we're worried about preserving the Drop behavior we could auto-call `stop()` in a manual `Drop` impl for `BackgroundProcessor`.